### PR TITLE
feat: add PaymentHintFlowUITests suite for payment-hint bottom sheet

### PR DESCRIPTION
**Pull request by zladzeyka**
## Pull Request Description

Adds `PaymentHintFlowUITests` — a BrowserStack UI-automation suite (17 tests) for the payment-hint bottom sheet added in PP-3260 / PP-3261 / PP-3263. Uses two PDF fixtures shared with Android PP-3301 (`invoice_future_due.pdf`, `invoice_no_due_date.pdf`) and drives R1–R13 show / no-show scenarios through the existing `paymentDueHintThresholdDays` public SDK knob — no test-only injection seam.

[PP-3302]

**How:**
- **CaptureSDK** — widened `InfoBottomSheetViewController.headerLabel` and `descriptionLabel` from `private` to internal so `PaymentHintBottomSheetViewController` can attach state-scoped accessibility identifiers on the container, title, description, primary, and secondary CTAs. A nested `AccessibilityIdentifiers` enum owns those identifier strings; SonarCloud-compliant `switch` maps each state to a static factory on the identifier struct (mirrored in the UITests target).
- **Example app (AppDelegate)** — reads a new `-paymentDueHintThresholdDaysOverride <Int>` launch argument and assigns it to `GiniBankConfiguration.shared.paymentDueHintThresholdDays`. Also skips Firebase on UI-test runs (`-StartFromCleanState`) so the committed empty `API_KEY` doesn't trip `FIRInstallations` on launch.
- **UITests — file flow** — `runFlowToAnalysis(fileName:)` selects the fixture by exact name via `MainScreen.tapFileFromBestAvailableSource(fileName:)`. On BS the fixture lives in Files.app "Custom_Files" (populated by `uploadMedia` PDFs); on local sim it lives under On-My-iPhone → GiniBankSDKExample (populated by `copyFixturesToSimulator`). Chooses `uploadFilesButton` (Files-app picker) over `uploadPhotoButton` (Photos) so filename selection actually works. PDFs skip the review-screen `Process` button step (matches the Skonto Files flow).
- **UITests — timezone** — `FIXTURE_DUE_DATE` is parsed identically to how the SDK ingests `paymentDueDate` (device-default timezone) so `remainingDays()` and `Date.isDueSoon(within:)` agree regardless of the BS slot's timezone.
- **UITests — Transaction Docs alert** — `assertExtractionReachesMainScreen` dismisses the "Only for this transaction" alert if it appears (BrowserStack always shows it on first run) and verifies the SDK closes back to Main via the results-screen Done button.
- **Scripts** — `bs_run_payment_hint.sh` builds, uploads, and triggers the suite. Uploads PDFs via `upload_media` with `custom_id` matching each test's exact `fileName`. Trigger includes `singleRunnerInvocation: "true"` (one `xcodebuild` per device — big win on setup overhead) and the 3-device matrix below.
- **Spec** — `specs/PP-3302-feature.md` documents the R1–R13 mapping and design decisions.

## Notes for Reviewers

**Device / iOS matrix covered by BS:**
- iPhone 17 — iOS 26 (latest iPhone on iOS 26)
- iPhone 16 — iOS 18 (latest iPhone on iOS 18)
- iPhone 15 — iOS 17 (latest iPhone on iOS 17)

BS plan allows 2 parallel sessions + 5 queued (7 slots total); the trigger enqueues all three and 2 run concurrently.

**Manual checks (flows):**
- R1–R3: Due Date / Schedule sheet appearance at various threshold overrides (default, boundary, below)
- R5–R6: no-sheet paths (flags off, no `paymentDueDate` in extraction)
- R7 / R10: "Proceed Anyway" → results screen → SDK closes to Main
- R8: "Cancel Transfer" returns to Main
- R9: Schedule CTA fires the host-app schedule callback alert
- R11: backdrop tap does not dismiss either sheet
- R12: capture-suggestions banner is suppressed while the sheet is up
- R13: sheet elements remain present at `UICTContentSizeCategoryAccessibilityXXXL`

**Unit tests added:**
- 4 `@Test` methods in `PaymentHintBottomSheetViewControllerTests.swift` verifying the accessibility identifiers propagate through the view hierarchy for both DueDate and Schedule states.

**Screenshots / recordings:** none — this PR adds automation for existing UI, no visual changes.

**Latest BS verification:** 17/17 tests pass on iPhone 17 iOS 26.5 with the current commits (`singleRunnerInvocation` + PDF/Files-app flow). Cross-device (iOS 17 & 18) coverage was added on top of that green baseline — will be re-verified before merge.

**Follow-ups / known limitations:**
- R13 `assertNoTruncation` intentionally asserts only element existence at AXXXL, not hittability. The sheet's `contentScrollView` is expected to overflow at accessibility text sizes; requiring immediate `isHittable` on the pinned CTAs would flag that expected behavior as a bug.
- `stubbedPaymentDueDate` was considered and rejected per the spec — the pre-existing `paymentDueHintThresholdDays` public SDK knob is enough.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[PP-3302]: https://ginis.atlassian.net/browse/PP-3302?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/Scripts/bs_shared.sh [hunk 1]
```diff
@@ -47,7 +48,7 @@ DEVICE_1="iPhone 17-26"
 DEVICE_2="iPhone 16-18"
 
 # ── BrowserStack project ──────────────────────────────────────────────────────
-BS_PROJECT="GiniBankSDK-LiquidGlass-4.3.0"
+BS_PROJECT="GiniBankSDK-iOS-PaymentHints-4.5.0"
```
[the remark this answers is not in the document]
Comment (zladzeyka) [reply BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/Scripts/bs_shared.sh]: Added similar as we have done for     bs_run_ra)           BUILD_LABEL="ReturnAssistant" ;;

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+
+    func testDueDateProceedAnywayContinuesToExtraction() throws {
+        relaunchApp(thresholdOverride: 5)
+        setPaymentHintFlags(dueDate: true, schedule: false)
+        runFlowToAnalysis(fileName: TestFixtures.Files.invoiceFutureDue)
+
+        let dueDateContainer = app.otherElements["paymentHint.dueDate.container"]
+        XCTAssertTrue(dueDateContainer.waitForExistence(timeout: 10),
+                      "Due date container should appear")
+        paymentHintScreen.dueDateProceedButton.tap()
```
[the remark this answers is not in the document]
Comment (zladzeyka) [reply BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift]: keep it

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+    private func assertNoTruncation(container: XCUIElement,
+                                    title: XCUIElement,
+                                    description: XCUIElement,
+                                    primary: XCUIElement,
+                                    secondary: XCUIElement) {
+        XCTAssertTrue(container.exists, "Sheet container must exist at AXXXL")
+        XCTAssertTrue(title.exists, "Title label must exist at AXXXL")
+        XCTAssertTrue(description.exists, "Description label must exist at AXXXL")
+        XCTAssertTrue(primary.exists, "Primary button must exist at AXXXL")
+        XCTAssertTrue(secondary.exists, "Secondary button must exist at AXXXL")
```
[the remark this answers is not in the document]
Comment (zladzeyka) [reply BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift]: keep it for now

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/PaymentHint/PaymentHintBottomSheetViewController.swift [hunk 4]
```diff
@@ -56,6 +59,46 @@ public final class PaymentHintBottomSheetViewController: InfoBottomSheetViewCont
...
+        let description: String
+        let primaryButton: String
+        let secondaryButton: String
+
+        static let dueDate = StateIdentifiers(
+            container: AccessibilityIdentifiers.DueDate.container,
+            title: AccessibilityIdentifiers.DueDate.title,
+            description: AccessibilityIdentifiers.DueDate.description,
+            primaryButton: AccessibilityIdentifiers.DueDate.proceedButton,
+            secondaryButton: AccessibilityIdentifiers.DueDate.cancelButton)
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/PaymentHint/PaymentHintBottomSheetViewController.swift]: [Suggestion] Can we fix indentation here?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/PaymentHint/PaymentHintBottomSheetViewController.swift [hunk 5]
```diff
@@ -56,6 +59,46 @@ public final class PaymentHintBottomSheetViewController: InfoBottomSheetViewCont
...
+
+        static let schedulePayment = StateIdentifiers(
+            container: AccessibilityIdentifiers.Schedule.container,
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/PaymentHint/PaymentHintBottomSheetViewController.swift]: [Suggestion] Can we fix indentation here?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/PaymentHint/PaymentHintBottomSheetViewController.swift [hunk 6]
```diff
@@ -138,6 +181,31 @@ extension PaymentHintBottomSheetViewController {
...
+        }
+
+        enum Schedule {
+            static let container = "paymentHint.schedule.container"
+            static let title = "paymentHint.schedule.title"
+            static let description = "paymentHint.schedule.description"
+            static let scheduleButton = "paymentHint.schedule.scheduleButton"
+            static let proceedButton = "paymentHint.schedule.proceedButton"
+        }
+    }
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/PaymentHint/PaymentHintBottomSheetViewController.swift]: [Suggestion] Let's use structs. We agreed that instead of enums, we should use structs from now on.
Also is this the style we want to have the identifiers? If yes we need to instruct claude to know this as well as struct vs enums.

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/TestFixtures.swift [hunk 7]
```diff
@@ -6,18 +6,13 @@
...
-
- - `Camera`: filenames **with** `.png` extension — injected via BrowserStack camera,
- or used as the lookup key when falling back to Files import locally
- (extension is stripped automatically by `provideImageAndReachReviewScreen`).
- Place BS media in `TestSamples/TestSamplesForBS/` and upload via `bs_build_and_upload.sh`.
+ - `Files`: names without extension — imported via the Files app (On My iPhone → Downloads).
+ - `Camera`: names with `.png` — injected via BrowserStack camera, or used as the Files-import
+   lookup key locally (extension stripped by `provideImageAndReachReviewScreen`). Place BS media
+   in `TestSamples/TestSamplesForBS/` and upload via `bs_build_and_upload.sh`.
  */
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/TestFixtures.swift:16]: [Clarification] What's the difference between what was and what is changed?  I can not figure it out.

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift [hunk 8]
```diff
@@ -0,0 +1,468 @@
...
+    /**
+     Encoded on `invoice_future_due.pdf` — validated against the real Gini API 2026-08-17.
+     Parses in the device's default timezone so `remainingDays()` matches the SDK's
+     `Date.isDueSoon(within:)` on UTC-configured BS devices.
+     */
+    private static let FIXTURE_DUE_DATE: Date = {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd"
+        formatter.locale = Locale(identifier: "en_US_POSIX")
+        return formatter.date(from: "2028-09-01")!
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift]: [Suggestion] Can we try to avoid using force unwrap here?

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift [hunk 9]
```diff
@@ -0,0 +1,468 @@
...
+        let formatter = DateFormatter()
+        formatter.dateFormat = "dd.MM.yyyy"
+        return formatter.string(from: FIXTURE_DUE_DATE)
+    }()
+
+    override var additionalLaunchArguments: [String] { ["-DisableReturnAssistant"] }
+
+    // MARK: - Helpers
+
+    private var paymentHintScreen: PaymentHintScreen!
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift]: [Clarification] Until now, this has been part of the `GiniBankSDKExampleUITests`. Why are we moving this into the UI tests class?

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift [hunk 10]
```diff
@@ -0,0 +1,468 @@
...
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        paymentHintScreen = PaymentHintScreen(app: app)
+    }
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift]: [Clarification] Is this really needed here? We already have one in GiniBankSDKExampleUITests. Normally, PaymentHintFlowUITests should have only tests if possible.

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift [hunk 11]
```diff
@@ -0,0 +1,468 @@
...
+private extension XCUIElement {
+    /**
+     Complement to `waitForExistence(timeout:)` — returns `true` if the element disappears in time.
+     */
+    func waitForNonExistence(timeout: TimeInterval) -> Bool {
+        let gonePredicate = NSPredicate(format: "exists == false")
+        let expectation = XCTNSPredicateExpectation(predicate: gonePredicate, object: self)
+        return XCTWaiter().wait(for: [expectation], timeout: timeout) == .completed
+    }
+}
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/PaymentHintFlowUITests.swift]: [Suggestion] This method should be moved somewhere where we can reuse it.

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/TestFixtures.swift [hunk 12]
```diff
@@ -55,6 +50,18 @@ enum TestFixtures {
         /// SEPA invoice with a future payment due date (triggers the Due Date hint).
         static let sepaDueDate = "sepa_due_date"
```
Comment (ValentinaIancu-Gini) [inline BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/TestFixtures.swift]: [Clarification] Why did we not used this invoice?

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/PaymentHint/PaymentHintBottomSheetViewController.swift [hunk 13]
```diff
@@ -56,6 +59,46 @@ public final class PaymentHintBottomSheetViewController: InfoBottomSheetViewCont
...
+        let ids: StateIdentifiers
+        switch state {
+        case .dueDate:         ids = .dueDate
+        case .schedulePayment: ids = .schedulePayment
+        }
+        view.accessibilityIdentifier = ids.container
+        headerLabel.accessibilityIdentifier = ids.title
+        descriptionLabel.accessibilityIdentifier = ids.description
+        buttonsViewContainer.primaryButton.accessibilityIdentifier = ids.primaryButton
+        buttonsViewContainer.secondaryButton.accessibilityIdentifier = ids.secondaryButton
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/PaymentHint/PaymentHintBottomSheetViewController.swift]: [suggestion] I think these identifiers should be part of the view model of the InfoBottomSheetViewModel. That's how we avoid exposing unnecessary elements so we can see the identifiers. And I'm not sure if we need specific identifiers per hint, but rather per UI component.

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/PaymentHintBottomSheetViewControllerTests.swift [hunk 14]
```diff
@@ -75,6 +75,39 @@ struct PaymentHintDueDateStateTests {
...
+        )
+        _ = sut.view
+        #expect(sut.view.accessibilityIdentifier == PaymentHintBottomSheetViewController.AccessibilityIdentifiers.DueDate.container,
+                "Container view must carry the .dueDate container identifier")
+    }
+
+    @Test("Title, description, primary and secondary carry Due Date identifiers")
+    @MainActor func labelsAndButtonsCarryDueDateAccessibilityIdentifiers() {
+        let sut = PaymentHintBottomSheetViewController(
+            state: .dueDate(formattedDueDate: "13.08.2026",
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/PaymentHintBottomSheetViewControllerTests.swift]: [Suggestion] Can we please fix indentation style here?

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/PaymentHintBottomSheetViewControllerTests.swift [hunk 15]
```diff
@@ -75,6 +75,39 @@ struct PaymentHintDueDateStateTests {
         #expect(!sut.shouldShowDragIndicator,
                 "Payment Hint sheet must not show a drag indicator — dismissal is CTA-driven only")
     }
+
+    @Test("Container view carries the Due Date accessibility identifier")
+    @MainActor func containerCarriesDueDateAccessibilityIdentifier() {
+        let sut = PaymentHintBottomSheetViewController(
+            state: .dueDate(formattedDueDate: "13.08.2026",
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/PaymentHintBottomSheetViewControllerTests.swift]: [Suggestion] Can we please fix indentation style here?

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/PaymentHintBottomSheetViewControllerTests.swift [hunk 16]
```diff
@@ -181,6 +214,39 @@ struct PaymentHintScheduleStateTests {
         #expect(!sut.shouldShowDragIndicator,
                 "Payment Hint sheet must not show a drag indicator — dismissal is CTA-driven only")
     }
+
+    @Test("Container view carries the Schedule accessibility identifier")
+    @MainActor func containerCarriesScheduleAccessibilityIdentifier() {
+        let sut = PaymentHintBottomSheetViewController(
+            state: .schedulePayment(formattedDueDate: "13.08.2026",
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/PaymentHintBottomSheetViewControllerTests.swift]: [Suggestion] Can we please fix indentation style here?

File: CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/PaymentHintBottomSheetViewControllerTests.swift [hunk 17]
```diff
@@ -181,6 +214,39 @@ struct PaymentHintScheduleStateTests {
...
+        )
+        _ = sut.view
+        #expect(sut.view.accessibilityIdentifier == PaymentHintBottomSheetViewController.AccessibilityIdentifiers.Schedule.container,
+                "Container view must carry the .schedulePayment container identifier")
+    }
+
+    @Test("Title, description, primary and secondary carry Schedule identifiers")
+    @MainActor func labelsAndButtonsCarryScheduleAccessibilityIdentifiers() {
+        let sut = PaymentHintBottomSheetViewController(
+            state: .schedulePayment(formattedDueDate: "13.08.2026",
```
Comment (ValentinaIancu-Gini) [inline CaptureSDK/GiniCaptureSDK/Tests/GiniCaptureSDKTests/PaymentHintBottomSheetViewControllerTests.swift]: [Suggestion] Can we please fix indentation style here?

**Review by ValentinaIancu-Gini — COMMENTED** [review]
@zladzeyka I finish reviewing this PR and left a couple of suggestions. Thank you!

File: BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/TestFixtures.swift (hunk 7 above)
Comment (zladzeyka) [reply BankSDK/GiniBankSDKExample/GiniBankSDKExampleUITests/TestFixtures.swift]: Instead of 3 lines :  
Central registry of all document fixtures used across UI tests.
Add new fixtures here when introducing new test invoices or images.
 Never hard-code filenames in individual test files.
 we have 2 :
 Central registry of document fixtures used across UI tests. Add new fixtures here
 instead of hard-coding filenames in individual tests.

**Review by ValentinaIancu-Gini — APPROVED** [review]
@zladzeyka Looks good thank you!
